packages/parser/src/language/sankey/sankey.langium

```diff
@@ -0,0 +1,46 @@
+grammar Sankey
+
+// common stuff in ../common/common.langium
+// can't import directly because of SANKEY_LINK_NODE parse rule
+interface Common {
+  accDescr?: string;
+  accTitle?: string;
+  title?: string;
+}
+
+fragment TitleAndAccessibilities:
+  ((accDescr=ACC_DESCR | accTitle=ACC_TITLE | title=TITLE) NEWLINE+)+
+;
+
+terminal NEWLINE: /\r?\n/;
+terminal ACC_DESCR: /accDescr(?:[\t ]*:[\t ]*[^\n\r]*?(?=%%)|\s*{[^}]*})|accDescr(?:[\t ]*:[\t ]*[^\n\r]*|\s*{[^}]*})/;
+terminal ACC_TITLE: /accTitle[\t ]*:[\t ]*[^\n\r]*?(?=%%)|accTitle[\t ]*:[\t ]*[^\n\r]*/;
+terminal TITLE: /title(?:[\t ]+[^\n\r]*?|)(?=%%)|title(?:[\t ]+[^\n\r]*|)/;
+
+hidden terminal WHITESPACE: /[\t ]+/;
+hidden terminal YAML: /---[\t ]*\r?\n[\s\S]*?---[\t ]*(?!.)/;
+hidden terminal DIRECTIVE: /[\t ]*%%{[\s\S]*?}%%\s*/;
+hidden terminal SINGLE_LINE_COMMENT: /[\t ]*%%[^\n\r]*/;
+
+// actual grammar
+interface Sankey extends Common {
+  links: SankeyLink[];
+  nodes: string[];
+}
+
+entry Sankey returns Sankey:
+  NEWLINE*
+  "sankey-beta"
+  (
+    NEWLINE* TitleAndAccessibilities links+=SankeyLink+
+    | NEWLINE+ links+=SankeyLink+
+  )
+;
+
+SankeyLink:
+  source=SANKEY_LINK_NODE SANKEY_LINK_COMMA target=SANKEY_LINK_NODE SANKEY_LINK_COMMA value=SANKEY_LINK_VALUE NEWLINE+
+;
+
+terminal SANKEY_LINK_COMMA: /,/;
+terminal SANKEY_LINK_VALUE returns number: /"(0|[1-9][0-9]*)(\.[0-9]+)?"|[\t ]*(0|[1-9][0-9]*)(\.[0-9]+)?/;
```

Comment:
It seems that we have [the same regexp here](https://github.com/mermaid-js/mermaid/pull/4799/files#diff-2800f172cbb1237161eccf8237ed566660474011c787e4e40c45f18a22eae117R9). If this is a limitation of languim that definitely brings a boilerplate to the code.